packages/compass-data-modeling/src/components/new-diagram-form.tsx

```diff
@@ -136,50 +140,79 @@ function SelectCollectionsStep({
-              // The already selected item was not shown to the user (using search),
-              // and we have to keep it selected.
-              return item ? item.selected : true;
-            }
-          );
+    <>
+      <FormFieldContainer>
+        <SearchInput
+          aria-label="Search collections"
+          value={searchTerm}
+          data-testid="new-diagram-search-collections"
+          onChange={(e) => {
+            setSearchTerm(e.target.value);
+          }}
+        />
+      </FormFieldContainer>
+      <FormFieldContainer>
+        <SelectList
+          className={selectListStyles}
+          items={filteredCollections.map((collName) => {
+            return {
+              id: collName,
+              selected: selectedCollections.includes(collName),
+              'data-testid': `new-diagram-collection-checkbox-${collName}`,
+            };
+          })}
+          label={{ displayLabelKey: 'id', name: 'Collection Name' }}
+          onChange={(items) => {
+            // When a user is searching, less collections are shown to the user
+            // and we need to keep existing selected collections selected.
+            const currentSelectedItems = selectedCollections.filter(
+              (collName) => {
+                const item = items.find((x) => x.id === collName);
+                // The already selected item was not shown to the user (using search),
+                // and we have to keep it selected.
+                return item ? item.selected : true;
+              }
+            );
 
-          const newSelectedItems = items
-            .filter((item) => {
-              return item.selected;
-            })
-            .map((item) => {
-              return item.id;
-            });
-          onCollectionsSelect(
-            Array.from(new Set([...newSelectedItems, ...currentSelectedItems]))
-          );
-        }}
-      ></SelectList>
-    </FormFieldContainer>
+            const newSelectedItems = items
+              .filter((item) => {
+                return item.selected;
+              })
+              .map((item) => {
+                return item.id;
+              });
+            onCollectionsSelect(
+              Array.from(
+                new Set([...newSelectedItems, ...currentSelectedItems])
+              )
+            );
+          }}
+        ></SelectList>
+      </FormFieldContainer>
+      {showAutoInferOption && (
+        <FormFieldContainer>
+          <Checkbox
+            checked={automaticallyInferRelationships}
+            onChange={(evt) => {
+              onAutomaticallyInferRelationshipsToggle(
+                evt.currentTarget.checked
+              );
+            }}
+            label="Automatically infer relationships"
+            // @ts-expect-error Element is accepted, but not typed correctly
+            description={
+              <>
+                Using a special algorithm, Compass will try to automatically
```

Comment:
Minor nits

"Using a special algo" can be removed. This might just cause people to ask for details on that algo